public/apps/configuration/panels/audit-logging/audit-logging-general-settings.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiGlobalToastList,
+  EuiPageHeader,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { cloneDeep, set } from 'lodash';
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+import { AppDependencies } from '../../../types';
+import { EditSettingGroup } from './edit-setting-group';
+import { SETTING_GROUPS } from './constants';
+import { API_ENDPOINT_AUDITLOGGING } from '../../constants';
+import { updateAuditLogging } from '../../utils/audit-logging-view-utils';
+import { AuditLoggingSettings } from './types';
+import { buildHashUrl } from '../../utils/url-builder';
+import { ResourceType } from '../../types';
+
+export function AuditLoggingEditGeneralSetting(props: AppDependencies) {
+  const [editConfig, setEditConfig] = useState<AuditLoggingSettings>({});
```

Comment:
Open discussion, no immediate action needed.

@mihirsoni @seraphjiang 
I have a little concern here with a big objects containing dozens of fields that are all mutable. From my understanding, one field updated will trigger all the fields re-render, is that correct? Will that be an concern?

Reply:
Got it, thanks Mihir @mihirsoni .